app/react/Metadata/helpers/validator.js

```diff
@@ -12,22 +14,70 @@ export function notEmpty(val) {
   return !!val && val.trim() !== '';
 }
 
+export function labelAndUrl(val) {
+  return (
+    !notEmpty(val) ||
+    (notEmpty(val.label) && notEmpty(val.url)) ||
+    (!notEmpty(val.label) && !notEmpty(val.url))
+  );
+}
```

Comment:
it seems this is the same function as on the server metadata validation isValidLinkField ? but different  implementations ? we should make an effort to have at least the overlapping validation functions written only once, another example is notEmpty on the client vs isEmpty on the server.